src/epcpm/parameterstobitfieldsc.py

```diff
@@ -0,0 +1,504 @@
+            "",
+            "",
+        ]
+
+        include_guard = self.h_path.name.replace(".", "_").upper()
+
+        h_lines = [
+            f"#ifndef {include_guard}",
+            f"#define {include_guard}",
+            f"",
+            f'#include "interface.h"',
+            f"",
+            f"",
+        ]
+
+        if not self.skip_output:
+            for parameter_uuid, modbus_nodes in parameter_uuid_to_modbus_node.items():
+                members_interface_c_lines = []
+                members_info_c_lines = []
+
+                name_uuid = epcpm.pm_helper.convert_uuid_to_variable_name(
+                    parameter_uuid
+                )
+
+                # Generate the SunSpec related .c/.h rows.
+                builder = builders.wrap(
+                    wrapped=modbus_nodes["sunspec"],
+                    parameter_uuid_finder=self.sunspec_model.node_from_uuid,
+                )
+                more_c_lines, more_h_lines = builder.gen()
+                c_lines.extend(more_c_lines)
+                h_lines.extend(more_h_lines)
+
+                common_c_lines = []
+                # Generate C lines for SunSpec node
+                if "sunspec" in modbus_nodes:
+                    builder = builders.wrap(
+                        wrapped=modbus_nodes["sunspec"],
+                        parameter_uuid_finder=self.sunspec_model.node_from_uuid,
+                    )
+                    common_c_lines.extend(builder.gen_common_interface())
+                    members_interface_c_lines.extend(
+                        builder.gen_bitfield_members_interface()
+                    )
+                    members_info_c_lines.extend(builder.gen_members_interface())
+                else:
+                    common_c_lines.extend(builder.gen_default_common_interface())
+
+                # Generate C lines for static modbus node
+                if "staticmodbus" in modbus_nodes:
+                    builder = builders.wrap(
+                        wrapped=modbus_nodes["staticmodbus"],
+                        parameter_uuid_finder=self.staticmodbus_model.node_from_uuid,
+                    )
+                    common_c_lines.extend(builder.gen_common_interface())
+                    members_interface_c_lines.extend(
+                        builder.gen_bitfield_members_interface()
+                    )
+                    members_info_c_lines.extend(builder.gen_members_interface())
+                else:
+                    common_c_lines.extend(builder.gen_default_common_interface())
+
+                c_lines.extend(members_interface_c_lines)
+
+                # Generate the combined interface item code
+                c_lines.extend(
+                    [
+                        f"InterfaceItem_Bitfield interfaceItem_{name_uuid} = {{",
+                        [
+                            f".common = {{",
+                            [ccl for ccl in common_c_lines],
```

Comment:
rename ccl to common\_c\_line ?

Reply:
I was considering going the other way and changing the code like this:
`[c for c in common_c_lines],`

If I change `ccl` to `common_c_line`, black reformats this one liner to four lines. It's already a somewhat difficult to read code section and increasing the variable name length makes it worse, IMO.

Also note that this one liner is a common python syntax called "list comprehension".

Reply:
sounds good.  yea list comprehension is confusing